frontend/src/metabase/lib/utils.js

```diff
@@ -163,8 +169,27 @@ const MetabaseUtils = {
         .map(c => SPECIAL_COMPONENTS[c] || parseInt(c, 10));
     // [1, 2, 3, -2, 1]
 
-    const aComponents = getComponents(aVersion);
-    const bComponents = getComponents(bVersion);
+    const aComponentsSplit = aVersion.split("-").map(getComponents);
```

Comment:
`aComponentsSplit[0]` represents "version number" and `aComponentsSplit[1]` represents "label" (see https://semver.org/).
I suggest to use these names rather than `0` & `1` indexes.
This comment applies to `bComponentsSplit` as well.

```suggestion
    const [aVersionNumber, aLabel] = aVersion.split("-").map(getComponents);
```

This will also allow us to define default value for `aLabel` instead of using ad-hoc `?? []` later on.


```suggestion
    const [aVersionNumber, aLabel = []] = aVersion.split("-").map(getComponents);
```